raiden/src/raiden.ts

```diff
@@ -175,6 +184,22 @@ export class Raiden {
       ),
     );
 
+    this.transfers$ = state$.pipe(
+      map(state => state.sent),
+      distinctUntilChanged(),
+      concatMap(sent => from(Object.entries(sent))),
+      /* this scan stores a reference to each [key,value] in 'acc', and emit as 'changed' iff it
+       * changes from last time seen. It relies on value references changing only if needed */
+      scan<[string, SentTransfer], { acc: SentTransfers; changed?: SentTransfer }>(
+        ({ acc }, [k, v]) => (acc[k] === v ? { acc } : { acc: { ...acc, [k]: v }, changed: v }),
```

Comment:
While I understand exactly what this line does. I had to pause for a minute or two to read it :smile:

Reply:
Haha, sorry, I'll try to make it more explicit.. is just that those mappers/reducers fit nicely on a single line, but sometimes can be harder to understand.